torch/fx/proxy.py

```diff
@@ -112,17 +112,47 @@ def __init__(self, graph: Graph):
 class TraceError(ValueError):
     pass
 
-# Proxy objects are stand-in values for normal values in a PyTorch computation.
-# Instead of performing compute they record computation into Graph.
-# Each proxy wraps the Node instance that represents the expression that define the
-# value.
 
 class Proxy:
+    """
+    ``Proxy`` objects are ``Node`` wrappers used by the ``Tracer`` to
+    record operations seen during symbolic tracing.
+
+    The mechanism through which ``Proxy`` objects record computation is
+    ``__torch_function__``. If any custom Python type defines a method
+    named ``__torch_function__``, PyTorch will invoke that
+    ``__torch_function__`` implementation when an instance of that
+    custom type is passed to a function in the ``torch`` namespace. In
+    the ``Proxy`` implementation of ``__torch_function__``, we simply
+    create a new ``Proxy`` object with the passed-in operator. Creating
+    a new ``Proxy`` means that a new ``Node``—the ``Node`` wrapped by
+    the ``Proxy``—will be added to the ``Graph`` as well. When this new
+    ``Proxy`` object is used later in the code, it “transforms” its user
+    into another ``Proxy`` object and the process is repeated.
+
+    Consider the following example:
+
+    .. code-block:: python
+
+        class M(torch.nn.Module):
+            def forward(self, x):
+                return torch.relu(x)
+
+    First, the parameter ``x`` is transformed into a ``Proxy`` object
```

Comment:
Probably want to have a line in the example showing a callsite for `symbolic_trace` or `Tracer.trace`. As it is, it's not clear who's doing this transformation